engine/src/main/java/org/pentaho/di/repository/RepositoryConnectionUtils.java

```diff
@@ -0,0 +1,178 @@
+/*!
+ * This program is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License, version 2.1 as published by the Free Software
+ * Foundation.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along with this
+ * program; if not, you can obtain a copy at http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ * or from the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ * Copyright (c) 2019 Hitachi Vantara.  All rights reserved.
+ */
+
+package org.pentaho.di.repository;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.pentaho.di.core.Const;
+import org.pentaho.di.core.exception.KettleException;
+import org.pentaho.di.core.exception.KettleSecurityException;
+import org.pentaho.di.core.logging.LoggingBuffer;
+import org.pentaho.di.core.plugins.PluginRegistry;
+import org.pentaho.di.core.plugins.RepositoryPluginType;
+import org.pentaho.di.i18n.BaseMessages;
+import org.pentaho.platform.api.engine.ActionExecutionException;
+import org.pentaho.platform.api.engine.IAuthorizationPolicy;
+
+import java.io.ByteArrayInputStream;
+
+/**
+ *  Provides a utility method to connect to repository and return the repository object
+ */
+public class RepositoryConnectionUtils {
+  private static final Log logger = LogFactory.getLog( RepositoryConnectionUtils.class );
+  private static final String SINGLE_DI_SERVER_INSTANCE = "singleDiServerInstance";
+  private static Class<?> PKG = RepositoryConnectionUtils.class;
+
+
+  /**
+   * Connects to the PDI repository. If the  system settings "SINGLE_DI_SERVER_INSTANCE" is set, it will connect to
+   * repository using the embedded repository.xml otherwise it will connect to repository with the given name.
+   *
+   * @param  String RepositoryName
+   * @param LoggingBuffer
```

Comment:
Params in the javadoc don't match the actual parameters of the method

Reply:
I can fix that

Reply:
The correct javadoc syntax should the name of the param instead of the type:
```
   * @param repositoryName RepositoryName
   * @param isSingleDiServerInstance Checking if the property "singleDiServerInstance" in PentahoSystems is set
   * @param authorizationPolicy  AuthorizationPolicy
   * @param userName Repo user name
   * @param fullyQualifiedServerUrl Fully qualified server Url used for connecting to the repository
   * @param logBuffer
```
Also the first two have an extra space after **@param**